packages/migrate/src/commands/DbPull.ts

```diff
@@ -231,6 +256,9 @@ Then you can run ${chalk.green(getCommandWithExecutor('prisma db pull'))} again.
       } else if (e.code === 'P1012') {
         // Schema Parsing Error
         console.info() // empty line
+
+        // TODO: this error is misleading, as it gets thrown even when the schema is valid but the protocol of the given
+        // '--url' argument is different than the one written in the schema.prisma file.
```

Comment:
Here is how to reproduce:

- Initialize a new schema with e.g. the `mongodb` provider: `npx prisma init --datasource-provider mongodb`
- Ensure the schema is valid: `npx prisma validate`
- Run introspection with a URL having a different datasource protocol: `npx prisma db pull --url postgres://prisma:prisma@localhost:5432/tests`

You should see an error similar to the following:

```text
Prisma schema loaded from prisma/schema.prisma
Environment variables loaded from .env
Datasource "db"

✖ Introspecting

Error: P1012 Introspection failed as your current Prisma schema file is invalid

Please fix your current schema manually, use prisma validate to confirm it is valid and then run this command again.
Or run this command with the --force flag to ignore your current schema and overwrite it. All local modifications will be lost.
```

Reply:
Ok, then my analysis would be that the combination of `provider=mongodb` and the `--url` value lead to validation failing before Introspeciton is run. (Similar how setting an env var with a mismatched `protocol` would lead to a similar problem)

Reply:
Can you open an issue for this TODO as well pls?

Reply:
https://github.com/prisma/prisma/issues/13076